src/main/java/com/dev/cinema/controllers/OrderController.java

```diff
@@ -0,0 +1,58 @@
+package com.dev.cinema.controllers;
+
+import com.dev.cinema.model.ShoppingCart;
+import com.dev.cinema.model.Ticket;
+import com.dev.cinema.model.User;
+import com.dev.cinema.model.dto.OrderRequestDto;
+import com.dev.cinema.model.dto.OrderResponseDto;
+import com.dev.cinema.service.OrderService;
+import com.dev.cinema.service.ShoppingCartService;
+import com.dev.cinema.service.UserService;
+import com.dev.cinema.service.mappers.OrderMapper;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/orders")
+public class OrderController {
+    private final OrderService orderService;
+    private final UserService userService;
+    private final ShoppingCartService shoppingCartService;
+    private final OrderMapper orderMapper;
+
+    @Autowired
+    public OrderController(OrderService orderService,
+                           UserService userService,
+                           ShoppingCartService shoppingCartService,
+                           OrderMapper orderMapper) {
+        this.orderService = orderService;
+        this.userService = userService;
+        this.shoppingCartService = shoppingCartService;
+        this.orderMapper = orderMapper;
+    }
+
+    @PostMapping("/complete")
+    public void complete(@RequestBody OrderRequestDto orderRequestDto) {
```

Comment:
it seems to me we need Userid to complete order - why would you pass so many details on this request?

Reply:
This is in case my business logic expands.